astropy/time/time_helper/function_helpers.py

```diff
@@ -0,0 +1,46 @@
+"""
+Helpers for overriding numpy functions in
+`~astropy.time.Time.__array_function__`.
+"""
+import numpy as np
+
+from astropy.units.quantity_helper.function_helpers import FunctionAssigner
+
+# TODO: Fill this in with functions that don't make sense for times
+UNSUPPORTED_FUNCTIONS = {}
+# Functions that take two Time arguments and return a single Time
+TWOARG_HELPERS = {}
+
+function_helper = FunctionAssigner(TWOARG_HELPERS)
+
+
+@function_helper(helps={np.linspace})
+def two_arg(t1, t2, *args):
+    """
+    Parameters
+    ----------
+    t1, t2 :
+        Time-like arguments given to numpy function. At least one of these is
+        an `astropy.time.Time`, and the other should be castable to `Time`.
+    *args :
+        Additional arguments given to the numpy function.
+
+    Returns
+    -------
+    t : astropy.time.Time
+        One of the times, used to keep track of the time scale, location, and
+        format.
+    args : tuple
+        args to pass to the numpy function. The first two are ``t1.jd, t2.jd``.
+    """
+    from astropy.time import Time
+    if isinstance(t1, Time):
+        t = t1
+        t2 = Time(t2)
+        t2 = getattr(t2, t.scale)
+    else:
+        t = t2
+        t1 = Time(t1)
+        t1 = getattr(t1, t.scale)
```

Comment:
I think perhaps we're being too generous in trying to take care also of non-`Time` instances. My sense would be to start by insisting both arguments are `Time` - otherwise, for something like `np.linspace(t1, '2000-01-01')`, what scale are you going to use?

Two small complications:
1. It should also be possible to have two `TimeDelta` instances...
2. If you do not have 2 Time instances, you should return `NotImplemented` - the other argument might know what to do with a `Time`...

Reply:
👍  - I'm going to limit scope of this PR to just `Time` (not `TimeDelta`), but good to keep that in mind.